core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.metastore.rocks;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.master.metastore.BlockStore;
+import alluxio.util.CommonUtils;
+import alluxio.util.io.FileUtils;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.primitives.Longs;
+import org.apache.commons.io.Charsets;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.ColumnFamilyOptions;
+import org.rocksdb.DBOptions;
+import org.rocksdb.ReadOptions;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksIterator;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Block store backed by RocksDB.
+ */
+public class RocksBlockStore implements BlockStore {
+  private final String mBaseDir;
+
+  private String mDbPath;
+  private RocksDB mDb;
+  private ColumnFamilyHandle mDefaultColumn;
+  private ColumnFamilyHandle mBlockMetaColumn;
+  private ColumnFamilyHandle mBlockLocationsColumn;
+
+  /**
+   * Creates and initializes a rocks block store.
+   */
+  public RocksBlockStore() throws RocksDBException {
+    mBaseDir = Configuration.get(PropertyKey.MASTER_METASTORE_DIR);
+    RocksDB.loadLibrary();
+    initDb(mBaseDir);
+  }
+
+  @Override
+  public Optional<BlockMeta> getBlock(long id) {
+    byte[] length;
+    try {
+      length = mDb.get(mBlockMetaColumn, Longs.toByteArray(id));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+    if (length == null) {
+      return Optional.empty();
+    }
+    return Optional.of(new BlockMeta(id, Longs.fromByteArray(length)));
+  }
+
+  @Override
+  public void putBlock(BlockMeta meta) {
+    try {
+      mDb.put(mBlockMetaColumn, Longs.toByteArray(meta.getId()),
+          Longs.toByteArray(meta.getLength()));
```

Comment:
this is a manual "encoding" of the metadata. Will this be scalable for the future?